acstools/polarization_tools.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python
+"""
+Toolkit for analyzing HST/ACS polarization data.
+
+CHANGE LOG
+----------
+2020-09-22: Creation. (T. Desjardins)
+"""
+
+import numpy as np
+import os
+import yaml
+
+from astropy.table import Table
+
+
+class PolarizerTables:
+    """
+    A simple class for holding all of the polarization tables (as astropy tables) in attributes.
+    """
+    def __init__(self):
+        with open(os.path.normpath(os.path.join(os.path.dirname(__file__), os.pardir, 'data',
+                                                'polarizer_tables.yaml')), 'r') as yf:
+            self.table_data = yaml.safe_load(yf)
+
+        self.wfc_transmission = Table(self.table_data['transmission']['wfc'],
+                                      names=('filter', 't_para', 't_perp', 'correction'))
+
+        self.hrc_transmission = Table(self.table_data['transmission']['hrc'],
+                                      names=('filter', 't_para', 't_perp', 'correction'))
+
+        self.wfc_efficiency = Table(self.table_data['efficiency']['wfc'],
+                                    names=('filter', 'pol0', 'pol60', 'pol120'))
+
+        self.hrc_efficiency = Table(self.table_data['efficiency']['hrc'],
+                                    names=('filter', 'pol0', 'pol60', 'pol120'))
+
+
+class Polarization:
+    """
+    Class for handling ACS polarization data.
+
+    Examples
+    --------
+
+    >>> from acstools.polarization_tools import Polarization
+    >>> vela_181 = Polarization(63684, 67420, 63752, 'F606W', 'WFC', 348.084)
+    >>> vela_181.calc_stokes()
+    >>> vela_181.calc_polarization()
```

Comment:
A class makes sense if the methods can be called again and again with different inputs without having to go through the constructor again. But since your methods take no arguments, I wonder if a class is really necessary here. Should `Polarization` be re-designed into a function instead, which returns output `dict` mapping the polarization property names to values?

Reply:
I'm going to stick with the class approach, as the class is mostly a convenience to keep all the information packaged together in attributes, as well as to automatically grab the correct efficiencies and leak corrections from the table data. But I have broken out the hidden methods to independent functions so that users can use those instead of the class.